### PR TITLE
Remove JvmPauseMonitorTest#interruptOnStop test

### PR DESCRIPTION
**Pull request by ZacBlanco**
Mockito was recently upgraded which includes a new feature that allows us to mock final methods and classes. I rewrote the test to be simpler using one of Mockito's Spy classes.

### CODE REVIEW

File: core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (ZacBlanco) [inline core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: In reference to why this is needed: https://github.com/mockito/mockito/wiki/What%27s-new-in-Mockito-2#mock-the-unmockable-opt-in-mocking-of-final-classesmethods

File: core/common/src/test/java/alluxio/util/JvmPauseMonitorTest.java [hunk 2]
```diff
@@ -49,28 +49,16 @@ public void pauseMonitorStartStopTest() {
...
-        fail("Failed to await on barrier");
-      }
-    });
-    t1.start();
+    JvmPauseMonitor mon = Mockito.spy(new JvmPauseMonitor(1000, 900000, 90000));
     mon.start();
-    barrier.await();
+    Thread pmThread = Whitebox.getInternalState(mon, "mJvmMonitorThread");
+    Thread spied = Mockito.spy(pmThread);
+    doThrow(InterruptedException.class).when(spied).join();
```
Comment (ZacBlanco) [inline core/common/src/test/java/alluxio/util/JvmPauseMonitorTest.java:56]: Thread#join is final

File: core/common/src/test/java/alluxio/util/JvmPauseMonitorTest.java [hunk 3]
```diff
@@ -14,9 +14,9 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-import static org.mockito.Matchers.any;
+import static org.mockito.ArgumentMatchers.any;
```
Comment (ZacBlanco) [inline core/common/src/test/java/alluxio/util/JvmPauseMonitorTest.java:17]: org.mockito.Matchers is now deprecated, so I moved to the newer class

File: core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (yuzhu) [reply core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: does this affect other tests? i saw several other tests failed. Could you take a look?

Comment (ZacBlanco) [reply core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Took a look. Latest commits should fix. I make use of the new Mockito#mockStatic methods in 3.4.x and above

Comment (ZacBlanco) [reply core/common/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: ok..well.. more issues arose. Latest commit fixes the failures of the last build

**Comment by yuzhu** [comment]
@ZacBlanco so the final decision is to remove it?

**Comment by ZacBlanco** [comment]
Yeah. I couldn't find a reliable way to test this without using Mockito's inline mock make, but that breaks all of our powermock tests.

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Pull request merged**